packages/ui/src/SelectField.tsx

```diff
@@ -181,6 +182,37 @@ export const SelectField = <V,>({
     }
   }, [menuPortalTarget])
 
+  const onChangeMultipleFn = React.useCallback(
+    (values: SelectFieldOption<V>[] | null) => {
+      ;(onChange as (newValue: V[] | null) => void)(values === null ? null : values.map(({ value }) => value))
+    },
+    [onChange],
+  )
+
+  const onChangeFn = React.useCallback(
+    (option: SelectFieldOption<V | null>) => {
+      ;(onChange as (newValue: V | null) => void)(option === null ? null : option.value)
+    },
+    [onChange],
+  )
+
+  const selectedOption: SelectFieldOption<V> | SelectFieldOption<V>[] | undefined = useMemo(() => {
+    if (isMulti) {
+      // if it's multi value, let's transform a value array to an array containing SelectFieldOptions
+      const valueArray = value as V[] | null
+      return (
+        valueArray?.map((val) => ({
+          value: val,
+          // we need to find instead of value: object mapping so that we can support objects as values
+          label: (options.find(({ value }) => value === val) as SelectFieldOption<V>).label,
+        })) ?? []
+      )
```

Comment:
Nit: We can remove these here. 

```suggestion
      return
        valueArray?.map((val) => ({
          value: val,
          // we need to find instead of value: object mapping so that we can support objects as values
          label: (options.find(({ value }) => value === val) as SelectFieldOption<V>).label,
        })) ?? []
```

Reply:
I'm working on adjusting this to a creatable option. I'll push it soon, can you please re-review afterwards?

Reply:
Sure, sry. Missed the `WIP`.